lib/jsdom/living/events/MouseEvent-impl.js

```diff
@@ -16,31 +16,43 @@ class MouseEventImpl extends UIEventImpl {
   }
   get pageX() {
     if (this._dispatchFlag) {
-      return 0;
+      return this._initialContainingBlockRelativeX;
     }
     const offset = wrapperForImpl(this.view)?.scrollX || 0;
     return offset + this.clientX;
   }
   get pageY() {
     if (this._dispatchFlag) {
-      return 0;
+      return this._initialContainingBlockRelativeY;
     }
     const offset = wrapperForImpl(this.view)?.scrollY || 0;
     return offset + this.clientY;
   }
   get offsetX() {
     if (this._dispatchFlag) {
-      return 0;
+      return this._paddingEdgeRelativeX;
     }
     return this.pageX;
   }
   get offsetY() {
     if (this._dispatchFlag) {
-      return 0;
+      return this._paddingEdgeRelativeY;
     }
     return this.pageY;
   }
 
+  constructor(globalObject, args, privateData) {
+    super(globalObject, args, privateData);
+
+    // We cache this during init so that we have a stable value during dispatch
+    const { scrollX = 0, scrollY = 0 } = wrapperForImpl(this.view) ?? {};
+    this._initialContainingBlockRelativeX = this.clientX + scrollX;
+    this._initialContainingBlockRelativeY = this.clientY + scrollY;
```

Comment:
I would call these fields simply `this._pageX`, `this._pageY`, `this._offsetX`, `this._offsetY`. These names are shorter, and yet more comprehensible and more to the point.